utils/data.cpp

```diff
@@ -0,0 +1,239 @@
+#include "data.hpp"
+
+namespace jbrush {
+
+// Constructor makes some simple assignments
+ShapeData::ShapeData() : currentDirectory(""){}
+
+Vector<Shape*> ShapeData::load()
+{
+    Vector<Shape*> shapes;  // Vector of shape pointers to be returned
+
+    // Bring up a panel for the user to select a directory
+    currentDirectory = QFileDialog::getOpenFileName(nullptr/*mainCanvas*/, "Please Choose a File", STARTING_DIRECTORY, "Text files (*.txt)");
+
+    // Make a file at the directory if it is non-empty
+    if(currentDirectory != "")
+    {
+        // Throws: file didn't open
+        try
+        {
+            // Get the file at the current directory
+            // and make a text stream to operate on it
+            QFile file(currentDirectory);
+            QTextStream qin(&file);
+            Shape* shapeExtracted;  // Shape extracted from the file
+
+            if(!file.open(QIODevice::ReadOnly))
+            {
+               // If the file didn't open, throw an exception
+               QString errorMessage;
+               QTextStream errorStream(&errorMessage);
+               errorStream << "File at " << currentDirectory << " was not opened properly";
+               throw GeneralException(errorMessage);
+            }
+
+            // Set integer base. Speeds things up a little for the text stream
+            qin.setIntegerBase(10);
+
+            // Push each shape extracted back into the vector until we get to the end of the file
+            while(!qin.atEnd())
+            {
+                // Throws: shape's type cannot be identified
+                try {
+                    shapeExtracted = extractShape(qin);
+                    shapes.push_back(shapeExtracted);
+                }
+                catch (GeneralException& exception) {
+                    qDebug() << exception.errorMessage();
+                    // Try to do something fancy to guarantee a recovery of the text stream
+                }
+            }
+
+            // Close the file when loading is finised
+            file.close();
+        }// end try
+        catch (GeneralException& exception){
+            qDebug() << exception.errorMessage();
+        }
+    }// end if (directory is non-empty)
+
+    return shapes;
+}
+
+Shape* ShapeData::extractShape(QTextStream& qin)
+{
+    QString shapeType;  // Name of the shape being extracted currently
+    QList<qint32> dimensionsList;   // List of dimensions of the shape
```

Comment:
We should be using the data type that shape uses, which are ``int``. This prevents unpredictable data loss when converting.